snuba/migrations/snuba_migrations/transactions/0002_transactions_onpremise_compatibility.py

```diff
@@ -0,0 +1,179 @@
+                table_name="transactions_local",
+                column=Column(
+                    "transaction_status", WithDefault(UInt(8), str(UNKNOWN_SPAN_STATUS))
+                ),
+                after="transaction_op",
+            ),
+            operations.AddColumn(
+                storage_set=StorageSetKey.TRANSACTIONS,
+                table_name="transactions_local",
+                column=Column("_tags_flattened", String()),
+                after="tags",
+            ),
+            operations.AddColumn(
+                storage_set=StorageSetKey.TRANSACTIONS,
+                table_name="transactions_local",
+                column=Column("_contexts_flattened", String()),
+                after="contexts",
+            ),
+            operations.AddColumn(
+                storage_set=StorageSetKey.TRANSACTIONS,
+                table_name="transactions_local",
+                column=Column("user_hash", Materialized(UInt(64), "cityHash64(user)")),
+                after="user",
+            ),
+            # The following columns were originally created as non low cardinality strings
+            operations.ModifyColumn(
+                storage_set=StorageSetKey.TRANSACTIONS,
+                table_name="transactions_local",
+                column=Column("transaction_name", LowCardinality(String())),
+            ),
+            operations.ModifyColumn(
+                storage_set=StorageSetKey.TRANSACTIONS,
+                table_name="transactions_local",
+                column=Column("release", LowCardinality(Nullable(String()))),
+            ),
+            operations.ModifyColumn(
+                storage_set=StorageSetKey.TRANSACTIONS,
+                table_name="transactions_local",
+                column=Column("dist", LowCardinality(Nullable(String()))),
+            ),
+            operations.ModifyColumn(
+                storage_set=StorageSetKey.TRANSACTIONS,
+                table_name="transactions_local",
+                column=Column("sdk_name", WithDefault(LowCardinality(String()), "''")),
+            ),
+            operations.ModifyColumn(
+                storage_set=StorageSetKey.TRANSACTIONS,
+                table_name="transactions_local",
+                column=Column(
+                    "sdk_version", WithDefault(LowCardinality(String()), "''")
+                ),
+            ),
+            operations.ModifyColumn(
+                storage_set=StorageSetKey.TRANSACTIONS,
+                table_name="transactions_local",
+                column=Column("environment", LowCardinality(Nullable(String()))),
+            ),
+            operations.AddColumn(
+                storage_set=StorageSetKey.TRANSACTIONS,
+                table_name="transactions_local",
+                column=Column("message_timestamp", DateTime()),
+                after="offset",
+            ),
+        ]
+
+    def backwards_local(self) -> Sequence[operations.Operation]:
+        return []
```

Comment:
Is this supposed to have a backwards migration (and what about the distributed ones)?

Reply:
This is a one off migration to return the transactions to the same state that should actually exist after 0001_transactions for people migrating from before the migration framework. The backwards operations would be no-op.